src/dropdown/dropdown.ts

```diff
@@ -93,12 +132,44 @@ export class NgbDropdown {
     }
   }
 
+  ngAfterContentInit() {
+    if (this.container === 'body') {
+      this._zoneSubscription = this._ngZone.onStable.subscribe(() => {
+        positionElements(
```

Comment:
I've got around 500 elements on page with dropdowns hidden inside them for when I click and this is causing quite an issue with performance as it's updating all of their positions all the time.
I fixed this locally by wrapping this in a check for open:
```
if (this._open) {
 ...
}
```

As that's only really when it needs to stick